third-party/linshare/src/main/java/org/apache/james/linshare/client/ByteArrayDecoder.java

```diff
@@ -0,0 +1,44 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.linshare.client;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+
+import feign.FeignException;
+import feign.Response;
+import feign.codec.Decoder;
+
+public class ByteArrayDecoder implements Decoder {
+    private final Decoder defaultDecoder;
+
+    ByteArrayDecoder(Decoder defaultDecoder) {
+        this.defaultDecoder = defaultDecoder;
+    }
+
+    @Override
+    public Object decode(Response response, Type type) throws IOException, FeignException {
+        if (type.getTypeName().equals("byte[]")) {
```

Comment:
the decoder Decoder.Default() could be a final field parameter named byteArrayDecoder.
I find it confusing right now to have a bunch of defaultDecoder, DecoderDefault...

If i understand correctly this class is not really a ByteArrayDecoder but add the byteArrayDecoding capability to the provided decoder. The naming could better express this.